### PR TITLE
[SW-1543] Expose driver if, port, port range and extra memory percent configuration for external H2O cluster

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: py/ai/h2o/sparkling/H2OConf.py [hunk 1]
```diff
@@ -437,6 +437,22 @@ def set_run_as_user(self, user):
         self._jconf.setRunAsUser(user)
         return self
 
+    def setExternalH2ODriverIf(self, iface):
```
Comment (jakubhava) [inline py/ai/h2o/sparkling/H2OConf.py]: On purpose I'm changing to camel case as later, I would like to start deprecating the snake case getters and setters on `H2OConf`. After the deprecation is done, we can start generating this configuration class

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala [hunk 2]
```diff
@@ -62,6 +62,10 @@ trait ExternalBackendConf extends SharedBackendConf {
   def kerberosPrincipal = sparkConf.getOption(PROP_EXTERNAL_KERBEROS_PRINCIPAL._1)
   def kerberosKeytab = sparkConf.getOption(PROP_EXTERNAL_KERBEROS_KEYTAB._1)
   def runAsUser = sparkConf.getOption(PROP_EXTERNAL_RUN_AS_USER._1)
+  def externalH2ODriverIf = sparkConf.getOption(PROP_EXTERNAL_DRIVER_IF._1)
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: WDYT about `sparkConf.getOption(PROP_EXTERNAL_DRIVER_IF._1, PROP_EXTERNAL_DRIVER_IF._2)`? I know that in both cases `None` is returned, but if 
```val PROP_EXTERNAL_DRIVER_IF = ("spark.ext.h2o.external.driver.if", None)``` specifies a default value...

File: doc/src/site/sphinx/configuration/configuration_properties.rst [hunk 3]
```diff
@@ -319,6 +319,24 @@ External backend configuration properties
...
++-------------------------------------------------------+----------------+-------------------------------------+
+| ``spark.ext.h2o.external.driver.port``                |  ``None``      | Port of mapper->driver callback     |
+|                                                       |                | interface. Default value means      |
+|                                                       |                | automatic detection.                |
++-------------------------------------------------------+----------------+-------------------------------------+
+| ``spark.ext.h2o.external.driver.port.range``          | ``None``       | Range portX-portY of mapper->driver |
+|                                                       |                | callback interface; eg:             |
+|                                                       |                | 50000-55000.                        |
++-------------------------------------------------------+----------------+-------------------------------------+
+| ``spark.ext.h2o.external.extra.memory.percent``       | ``10``         | extramempercent is a percentage of  |
```
Comment (mn-mikke) [inline doc/src/site/sphinx/configuration/configuration_properties.rst]: Is `extramempercent` mentioned somewhere in the documentation?

File: py/ai/h2o/sparkling/H2OConf.py [hunk 4]
```diff
@@ -437,6 +437,22 @@ def set_run_as_user(self, user):
...
+    def setExternalH2ODriverPort(self, port):
+        self._jconf.setExternalH2ODriverPort(port)
+        return self
+
+    def setExternalH2ODriverPortRange(self, portrange):
+        self._jconf.setExternalH2ODriverPortRange(portrange)
+        return self
+
+    def setExternalExtraMemoryPercent(self, memoryPercent):
+        self._jconf.etExternalExtraMemoryPercent(memoryPercent)
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/H2OConf.py]: `etExternalExtraMemoryPercent` -> `setExternalExtraMemoryPercent`

File: doc/src/site/sphinx/configuration/configuration_properties.rst (hunk 3 above)
Comment (jakubhava) [reply doc/src/site/sphinx/configuration/configuration_properties.rst]: Good catch, copy paste error. We can write the documentation as : `Specifies a percentage ..`, what do you think?

File: py/ai/h2o/sparkling/H2OConf.py (hunk 4 above)
Comment (jakubhava) [reply py/ai/h2o/sparkling/H2OConf.py]: thanks!

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala (hunk 2 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: If I understand the motivation, this is to ensure we are prepared for cases when we change the default value of the underlying option & forget to change the getter?

The idea makes sense, however the code `getOption` does not accept second argument and `get` methods does not accept Option as value. Given these ^, I would keep it as it is. WDYT?

File: doc/src/site/sphinx/configuration/configuration_properties.rst (hunk 3 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/configuration/configuration_properties.rst]: `This option is a percentage of ...` sounds good to me.

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala (hunk 2 above)
Comment (mn-mikke) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: Ok, let's leave it as it is.

File: doc/src/site/sphinx/configuration/configuration_properties.rst (hunk 3 above)
Comment (jakubhava) [reply doc/src/site/sphinx/configuration/configuration_properties.rst]: cool!

**Pull request merged**